### ISSUE TITLE
Conventions

### ISSUE THREAD

**Issue by castwide** [castwide/solargraph#199]
A `Convention` is a set of modifications to the code maps based on the currently open files. The first three supported conventions are Gemfile, Gemspec, and Rspec.

* Files named `Gemfile` expose the Bundler DSL (`gem`, `gemspec`, etc.)
* Files ending with `.gemspec` automatically require `rubygems` and understand yielded specs in `Gem::Specification.new` blocks.
* Files ending with `_spec.rb` expose the RSpec DSL.

Conventions are currently in the `master` branch. They might also serve as a component for improving Rails support.

**Comment by minkir014** [comment]
By saying

> Conventions are currently in the master branch. They might also serve as a component for improving Rails support.



Do you mean that you have implemented it but haven't updated the gem nor the editor's extension???

**Comment by castwide** [comment]
Yes. It'll be released in gem version 0.34 and won't require an extension update.

**Comment by castwide** [comment]
Released in 0.34.0.

**Comment by hellola** [comment]
Hey I've also been struggling getting the ActiveRecord to work well with solargraph. I'm not sure if I'm missing anything crucial. I've followed all the rails related help I could find.
I made a small generator using activerecord reflections, here is the [gist](https://gist.github.com/hellola/b832cff599f67a5e353e6af4914d467a) 
It just generates classes with yard / solargraph comments in them.
  @Zeko369  @tvallois  @qortex @castwide
